### PR TITLE
#8 :bug: activate pyenv and python before clean; #7 :arrow_up: bump p…

### PR DESCRIPTION
**Pull request by d-ryan-ashcraft**
…ython 3.9 to latest patch release

### CODE REVIEW

File: habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/PyenvAndPoetrySetup.java [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+ * <li>Required Poetry plugins (currently only {@code poetry-monorepo-dependency-plugin})</li>
+ * </ul>
+ */
+class PyenvAndPoetrySetup {
+
+    /**
+     * Specifies the semver compliant requirement for the default version of Python that
+     * must be installed and available for Habushu to use.
+     */
+    protected static final String PYTHON_DEFAULT_VERSION_REQUIREMENT = "3.9.16";
```
Comment (d-ryan-ashcraft) [inline habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/PyenvAndPoetrySetup.java:35]: I: this is the bump to 3.9.16 - now two places where Python can be instantiated, but both use this constant

File: habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/PyenvAndPoetrySetup.java [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+/**
+ * Common class that ensures pre-requisite tools that Habushu leverages are installed and available on the
+ * developer's machine to support the same functionality across multiple Mojo implementations. These include:
+ * <ul>
+ * <li>pyenv</li>
+ * <li>Poetry (installed version must satisfy {@link #POETRY_VERSION_REQUIREMENT})</li>
+ * <li>Required Poetry plugins (currently only {@code poetry-monorepo-dependency-plugin})</li>
+ * </ul>
+ */
+class PyenvAndPoetrySetup {
```
Comment (d-ryan-ashcraft) [inline habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/PyenvAndPoetrySetup.java:29]: I (informational): this is almost exclusively refactoring prior implementation for reuse in two plugins that cannot share a base class.  Thus we are using a composition approach.

File: habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/exec/PyenvCommandHelper.java [hunk 3]
```diff
@@ -25,29 +27,31 @@ public class PyenvCommandHelper {
...
     public boolean isPyenvInstalled() {
-	try {
-	    execute(Arrays.asList("--version"));
-	} catch (Throwable e) {
-	    return false;
-	}
-	return true;
+        try {
+            String foundVersion = executeWithDebugLogging(Arrays.asList("--version"));
+            logger.debug("Found " + foundVersion);
```
Comment (d-ryan-ashcraft) [inline habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/exec/PyenvCommandHelper.java:39]: I: just made some of the debugging with pyenv debug level so it's a more crisp in the info output what is happening that people typically care about

**Review by d-ryan-ashcraft — COMMENTED** [review]
Recommend ignoring whitespace

**Review by kdendtler — APPROVED** [review]

**Review by ekoniec1 — APPROVED** [review]

**Pull request merged**